From: Benjamin Otte <otte@redhat.com>
Date: Fri, 21 Nov 2025 10:07:31 +0100
Subject: vulkan: Don't treat Vulkan objects as pointers or 32bit breaks

Fixes: #7883
Part-of: <https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/9170>
(cherry picked from commit ca58e397dc3cc96027ea724be61de002f36064cb)
---
 gdk/gdkvulkancontext.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gdk/gdkvulkancontext.c b/gdk/gdkvulkancontext.c
index 541842b..2639389 100644
--- a/gdk/gdkvulkancontext.c
+++ b/gdk/gdkvulkancontext.c
@@ -688,7 +688,7 @@ gdk_vulkan_present_is_busy (GdkVulkanContext *self,
 {
   VkResult res;
 
-  if (present->vk_swapchain == NULL)
+  if (!present->vk_swapchain)
     return FALSE;
 
   if (!present->vk_fence)
@@ -1208,7 +1208,7 @@ gdk_vulkan_context_surface_detach (GdkDrawContext *context)
       if (priv->presents[i].vk_swapchain)
         {
           gdk_vulkan_context_unref_swapchain (self, priv->presents[i].vk_swapchain);
-          priv->presents[i].vk_swapchain = NULL;
+          priv->presents[i].vk_swapchain = VK_NULL_HANDLE;
         }
       vkDestroySemaphore (vk_device,
                           priv->presents[i].vk_semaphore,
